nix-ts-mode.el

```diff
@@ -151,18 +157,30 @@
    :language 'nix
    :feature 'function
    `((function_expression
-      ":" @font-lock-misc-punctuation-face)))
+      ":" @font-lock-misc-punctuation-face))
 
+   :language 'nix
+   :feature 'error
+   :override t
+   '((ERROR) @font-lock-warning-face))
   "Tree-sitter font-lock settings for `nix-ts-mode'.")
 
 ;; Indentation
 (defvar nix-ts-mode-indent-rules
-  (let ((offset nix-ts-mode-indent-offset))
-    `((nix
-       ((node-is "}") parent-bol 0)
-       ((node-is ")") parent-bol 0)
-       ((node-is "]") parent-bol 0)
-       ((parent-is "parenthesized_expression") parent-bol ,offset)))))
+  `((nix
+     ((parent-is "source_code") column-0 0)
+     ((node-is "]") parent-bol 0)
+     ((node-is ")") parent-bol 0)
+     ((node-is "}") parent-bol 0)
+     ((node-is "binding_set") parent-bol nix-ts-mode-indent-offset)
+     ((node-is "indented_string_expression") parent-bol nix-ts-mode-indent-offset)
```

Comment:
Just tried this branch & it does indeed indent much better than stock nix-ts-mode. However, I can't get indented string expressions to indent right - whenever I open a `''` string, it indents to column zero:

Given 
```nix
{
  bar = ''
# cursor <-
  '';
}
```
nix-ts-mode places insertion point at bol.

From the treesit-explore output of that,

```
expression: 
 (attrset_expression {
  (binding_set
   binding: 
    (binding
     attrpath: (attrpath attr: (identifier))
     =
     expression: (indented_string_expression '' (string_fragment) '')
     ;))
  })
```

It looks like this should be a string_fragment scoped to indented_string_expression... is that possible?

Reply:
@marsam I just [opened a PR](https://github.com/marsam/nix-ts-mode/pull/1) to your branch to partially fix this.
[Not perfect but better than nothing](https://asciinema.org/a/Tbb6oDw3uutGTDMPFUz1JJpNo)

Reply:
We can fix this one incrementally.